mllib/src/main/scala/org/apache/spark/ml/feature/LSH.scala

```diff
@@ -75,7 +75,7 @@ private[ml] abstract class LSHModel[T <: LSHModel[T]]
    * The hash function of LSH, mapping an input feature vector to multiple hash vectors.
    * @return The mapping of LSH function.
    */
-  protected[ml] val hashFunction: Vector => Array[Vector]
+  protected[ml] def hashFunction(elems: Vector): Array[Vector]
```

Comment:
This change does appear to resolve the issue by avoiding whatever is happening in these two cases. This is at least reasonable, as it sort of appears it's something to do with scala + Java 8 rather than Spark.

However I do wonder whether MiMa will allow this change. It is still exposed as a function in the bytecode, so maybe. If not, will be a tough call whether this experimental class is allowed to change.